### DISCUSSION TITLE
Is it possible to increase the session.gc_maxlifetime php.ini setting?

### DISCUSSION THREAD

**Discussion by KenBW2** [nextcloud/all-in-one#3673] [Questions] [answered]
I'm debugging a problem where I'm logged out of NC on a regular basis. I haven't found a consistent time, but it's on a daily basis and is really annoying

I note that the `session_lifetime` is limited by the `session.gc_maxlifetime` php.ini setting, which is the default 24 minutes

How do I go about increasing the `session.gc_maxlifetime` php.ini setting to see if that's the issue?

**Comment by szaimen** [comment]
Hi, in AIO we are using redis sessions: https://github.com/nextcloud/all-in-one/blob/ffe5ba2a1bd6ab08e1b997ec57b3075ab449f0c5/Containers/nextcloud/entrypoint.sh#L24

So I doubt that increasing this value will fix the problem. However you could temporarily change the valur inside the container and check if that actually improves things.

However I rather think that you are running into this: https://github.com/nextcloud/server/issues/37492

**Reply by ShawkMusic** [reply]
I saw #3674 which seems to describe exactly the problem I have. That post is locked and linked to here.

So... what exactly am I supposed to do?

**Comment by joshtrichards** [comment]
@szaimen The Redis session handler still utilizes the `session.gc_maxlifetime` value:

https://github.com/phpredis/phpredis/blob/0fe45d24d4d8c115a5b52846be072ecb9bb43329/redis_session.c#L133-L145

**Reply by szaimen** [reply]
Thanks for the link @joshtrichards! This explains things :)

**Comment by szaimen** [comment, marked as answer]
This is getting worked on in https://github.com/nextcloud/all-in-one/pull/5497

**Reply by szaimen** [reply]
This is now released with v9.8.0 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel
